### ISSUE TITLE
CUDA memory is not reduced when training ViT-B architecture on ImageNet 224x224 using the provided loss function

### ISSUE THREAD

**Issue by jiachenlei** [DAMO-NLP-SG/Inf-CLIP#5]
Hi, Thanks for your great work and open-sourcing your codes! 

I am working on one of my project that utilizes contrastive loss.  My code computes the contrastive loss for three times each with different input. I follow the instructions in README.md and simply replace code snippets that compute contrastive loss with inf_cl_loss. However, the CUDA memory usage is not reduced as expected.

In addition to this repo, I also utilized the *xformers* repository for efficient ViT attention computation and mixed precision (fp16) training.

Do you have any clues why this is happening? Thanks!

**Comment by jiachenlei** [comment]
I emailed the author and they mentioned that one should apply gradient cache (https://github.com/luyug/GradCache) together with the loss function in this repo.

**Comment by Pengxiang-zhao** [comment]
Could you please let me know which email you used to contact the author? I would like to reach out to the author as well. Thank you!
